src/snowflake/cli/plugins/nativeapp/project_model.py

```diff
@@ -105,16 +105,16 @@ def stage_schema(self) -> Optional[str]:
     @cached_property
     def package_warehouse(self) -> Optional[str]:
         if self.definition.package and self.definition.package.warehouse:
-            return self.definition.package.warehouse
+            return to_identifier(self.definition.package.warehouse)
         else:
-            return cli_context.connection.warehouse
+            return to_identifier(cli_context.connection.warehouse)
 
     @cached_property
     def application_warehouse(self) -> Optional[str]:
         if self.definition.application and self.definition.application.warehouse:
-            return self.definition.application.warehouse
+            return to_identifier(self.definition.application.warehouse)
         else:
-            return cli_context.connection.warehouse
+            return to_identifier(cli_context.connection.warehouse)
```

Comment:
`cli_context.connection.warehouse` might be `None`, in which case we don't want to pipe it through `to_identifier`.